feat(api-service): Nv 5304 update pricing and billing pages

tatarco opened the pull request:
### What changed? Why was the change needed?

<!-- Also include any relevant links, such as Linear tickets, Slack discussions, or design documents. -->

### Screenshots

<!-- If the changes are visual, include screenshots or screencasts. -->

<details>
<summary><strong>Expand for optional sections</strong></summary>

### Related enterprise PR

<!-- A link to a dependent pull request  -->

### Special notes for your reviewer

<!-- Specific instructions or considerations you want to highlight for the reviewer. -->

</details>

ChmaraX commented on apps/dashboard/src/hooks/use-checkout-session.ts:
```diff
@@ -20,14 +20,14 @@ export function useCheckoutSession() {
       post<CheckoutResponse>('/billing/checkout-session', {
         body: {
           billingInterval,
-          apiServiceLevel: ApiServiceLevelEnum.BUSINESS,
```
this needs to be behind a feature flag for now

ChmaraX commented on apps/dashboard/src/hooks/use-checkout-session.ts:
```diff
@@ -20,14 +20,14 @@ export function useCheckoutSession() {
         body: {
           billingInterval,
-          apiServiceLevel: ApiServiceLevelEnum.BUSINESS,
+          apiServiceLevel: ApiServiceLevelEnum.TEAM,
```
this needs to be behind a feature flag for now

tatarco replied:
maybe lets put the feature flag in the backend? WDYT? so it will be ready

ChmaraX commented on packages/shared/src/types/organization.ts:
```diff
@@ -1,7 +1,9 @@
 export type OrganizationId = string;
 
 export enum ApiServiceLevelEnum {
+  /** @deprecated Use TEAM instead */
```
use FREE_2025 instead

ChmaraX commented on apps/dashboard/src/components/billing/features.tsx:
```diff
@@ -1,272 +1,196 @@
+  suffix?: string | React.ReactNode;
+}
+
+function buildValues(params: BuildValuesParams): Record<SupportedPlansEnum, FeatureValue> {
+  return Object.values(SupportedPlansEnum).reduce(
+    (acc, plan) => {
+      const apiServiceLevel = SupportedPlansEnumToServiceLevelRecord[plan];
+      if (params.isBoolean) {
+        const bool = params.featureName ? getFeatureForTierAsBoolean(params.featureName, apiServiceLevel) : '';
+        acc[plan] = {
+          value: bool ? <Check className="h-4 w-4" /> : '-',
+        };
+        return acc;
+      }
+      const text = params.featureName ? getFeatureForTierAsText(params.featureName, apiServiceLevel) : '';
+      const value = `${params.prefix || ''}${text}${params.suffix || ''}`;
+      acc[plan] = {
+        value,
+      };
+      return acc;
+    },
+    {} as Record<SupportedPlansEnum, FeatureValue>
+  );
+}
+
```
This seems like an over-abstraction to me - I would simply leave the frontend plans hardcoded, there don't need to be tied to the shared tier config.

ChmaraX commented on apps/dashboard/src/components/billing/highlights-row.tsx:
```diff
@@ -1,33 +1,23 @@
-  ],
-  [ApiServiceLevelEnum.BUSINESS]: [
-    { text: 'Up to 250,000 events per month' },
-    { text: 'Unlimited teammates' },
-    { text: '90 days Activity Feed retention' },
-  ],
-  [ApiServiceLevelEnum.ENTERPRISE]: [
-    { text: 'Up to 5,000,000 events per month' },
-    { text: 'Unlimited teammates' },
-    { text: 'SAML SSO' },
-  ],
-};
+const highlights: PlanHighlights = Object.entries(serviceLevelToFuncArray).reduce(
+  (highlights, [serviceLevelKey, funcs]) => {
+    highlights[serviceLevelKey as ApiServiceLevelEnum] = funcs.map((func) => ({
+      text: func(serviceLevelKey as ApiServiceLevelEnum),
+    }));
+    return highlights;
+  },
+  {} as PlanHighlights
+);
```
Same as before, I would keep it hardcoded as it was - this requires significant mental overhead to understand

ChmaraX commented on apps/dashboard/src/components/billing/plans-row.tsx:
```diff
@@ -1,160 +1,197 @@
+  return `billed ${interval === 'year' ? 'annually' : 'monthly'}`;
+}
+
+function buildPlanConfig(
```
same as previous comment, abstraction tied too closely to the tier config, just let it be hardcoded so its very easy to understand and modify

tatarco replied:
I don't agree, it just shouldn't be a thing you need to manage, there should be a single source of truth, no one needs to do anything

ChmaraX reviewed (changes requested):
I like the idea of having a single place for the system limits!

But I would revert all the dashboard abstractions and leave them stupid simple as before, its easier to understand and easier to change, it also doesn't change together that often.

tatarco committed:
fix(dashboard): added FF to FE

fix(dashboard): added FF to FE

fix(dashboard): added FF to FE

fix(dashboard): change default sorting

fix(api-service): compiles68457b4c79403ba223d4f360925eb0422e33dffb

fix(api-service): compiles68457b4c79403ba223d4f360925eb0422e33dffb

fix(api-service): compiles

djabarovgeorge commented on .source:
does this PR has changes on EE repo? if it does please attach a link to the associated PR in this PR description,

tatarco replied:
it does, will do

tatarco commented:
testing and reopening
